core/startup/ErrorControlChain.php

```diff
@@ -36,9 +36,21 @@ public function setErrored($error) {
 		$this->error = (bool)$error;
 	}
 
+	/**
+	 * Sets whether errors are suppressed or not
+	 * Notes:
+	 * - Errors cannot be suppressed if not handling errors.
+	 * - Errors cannot be un-suppressed if original mode dis-allowed visible errors
+	 *
+	 * @param bool $suppression
+	 */
 	public function setSuppression($suppression) {
 		$this->suppression = (bool)$suppression;
-		if ($this->handleFatalErrors) ini_set('display_errors', !$suppression);
+		// Don't modify errors unless handling fatal errors, and if errors were
+		// originally allowed to be displayed.
+		if ($this->handleFatalErrors && $this->originalDisplayErrors) {
```

Comment:
I can't see where in the code `this->suppression` gets set to the value obtained from `display_errors`? Seems like it's always true or false.  My comment is related to line 51, where the second condition would always evaluated to true